x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx

```diff
@@ -88,6 +91,9 @@ export function TimeseriesChart({
   const { setPointerEvent, chartRef } = useChartPointerEventContext();
   const theme = useTheme();
   const chartTheme = useChartTheme();
+  const [selectedAlertId, setSelectedAlertId] = useState<string | undefined>(
```

Comment:
curious if this + <LazyAlertsFlyout> could be better as a hook? One that returns both the result from `getAlertAnnotations` and the `LazyAlertFlyout` component?

Reply:
Interesting. I'm doing some more work on the flyout in #96568 so I'll play with that idea when I work on it next.